app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php

```diff
@@ -58,24 +58,23 @@ public function __construct(
      *
      * @param Quote $cart
      * @param array $cartItemData
-     * @return void
-     * @throws GraphQlNoSuchEntityException
      * @throws GraphQlInputException
+     * @throws GraphQlNoSuchEntityException
      * @throws \Magento\Framework\Exception\LocalizedException
      */
     public function execute(Quote $cart, array $cartItemData): void
     {
         $sku = $this->extractSku($cartItemData);
         $qty = $this->extractQty($cartItemData);
-        $customizableOptions = $this->extractCustomizableOptions($cartItemData);
 
         try {
             $product = $this->productRepository->get($sku);
         } catch (NoSuchEntityException $e) {
             throw new GraphQlNoSuchEntityException(__('Could not find a product with SKU "%sku"', ['sku' => $sku]));
         }
-
+      
         try {
+            $customizableOptions = $this->extractCustomizableOptions($product, $cartItemData);
             $result = $cart->addProduct($product, $this->createBuyRequest($qty, $customizableOptions));
         } catch (\Exception $e) {
             throw new GraphQlInputException(
```

Comment:
We should not show all of exception types to client.
We need to split on Localized and the other errors.